website/js/index.js

```diff
@@ -588,15 +705,196 @@ var VariantDetail = React.createClass({
+                                <thead>
+                                    <tr className='active'>
+                                        <th>Release Date</th>
+                                        <th>Clinical Significance</th>
+                                        <th>Changes</th>
+                                    </tr>
+                                </thead>
+                                <tbody>
+                                    {diffRows}
+                                </tbody>
+                            </Table>
+                            <p style={{display: this.props.mode === "research_mode" ? 'none' : 'block' }}>There may be additional changes to this variant, click "Show All Public Data on this Variant" to see these changes.</p>
+                        </Col>
+                    </Row>
+                    <Row>
+                        <Col md={8} mdOffset={2}>
+                            <DisclaimerModal buttonModal onToggleMode={this.onChildToggleMode} text="Show All Public Data on this Variant"/>
+                        </Col>
+                    </Row>
+                </Grid>
+        );
+    },
+    renderAllData: function () {
+        const {data, error} = this.state;
+        if (!data) {
+            return <div />;
+        }
+
+        let variant = data[0],
+            release = variant["Data_Release"],
+            cols;
+
+        // FAISAL: this method is now specific to research mode (aka the all data portal), so no need to check local storage
+        cols = researchModeColumns;
+
+        // FAISAL: rather than directly map cols, we create a higher-level groups structure
+        // the higher-level groups structure maps a subset of columns to that group
+        let groupsEmpty = 0;
+        let totalRowsEmpty = 0;
+
+        const groupTables = _.map(groups, ({groupTitle, innerCols }) => {
+            let rowsEmpty = 0;
+
+            // now map the group's columns to a list of row objects
+            const rows = _.map(innerCols, ({prop, title}) => {
+                let rowItem;
+
+                if (prop === "Protein_Change") {
+                    title = "Abbreviated AA Change";
+                }
+                if (variant[prop] !== null) {
+                    if (prop === "Gene_Symbol") {
+                        rowItem = <i>{variant[prop]}</i>;
+                    }
+                    else if (prop === "URL_ENIGMA") {
+                        if (variant[prop].length) {
+                            rowItem = <a target="_blank" href={variant[prop]}>link to multifactorial analysis</a>;
+                        }
+                    } else if (prop === "Assertion_method_citation_ENIGMA") {
+                        rowItem = <a target="_blank" href="https://enigmaconsortium.org/library/general-documents/">Enigma Rules version Mar 26, 2015</a>;
+
+                        // this will be used in All Data display
```

Comment:
apologies, forgot to respond to this one. i didn't write that code, so i was reluctant to remove it, but sure, i don't think it's doing anything commented-out in there aside from making block even more difficult to read. i'll go ahead and remove it and otherwise fix up the formatting there.